src/tree.rs

```diff
@@ -164,12 +175,12 @@ impl<ObjectID: FsVerityHashValue> Directory<ObjectID> {
                 Component::Normal(filename) => match dir.entries.get(filename) {
                     Some(Inode::Directory(subdir)) => subdir,
                     Some(_) => return Err(ImageError::NotADirectory(filename.into())),
-                    None => return Err(ImageError::NotFound(filename.into())),
+                    None => return Ok(None),
```

Comment:
aww.  I expected to see some fancy `.map().transpose()` work here ;)

Reply:
er.  I wrote this comment int he wrong place.  I meant it for the get_file_opt() case, sorry.